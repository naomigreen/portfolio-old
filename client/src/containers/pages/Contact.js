import React, { useState } from 'react'
import styled from 'styled-components';
import { useForm } from 'react-hook-form';
import Form from '../../components/Form/Form';

export default function Contact() {
	const { register, handleSubmit, reset } = useForm();
	const [sending, setSending] = useState(false);

	const onSubmit = data => {
		setSending(true)
		sendMessage(
			data.name,
			data.email,
			data.message,
			process.env.REACT_APP_EMAILJS_TEMPLATEID,
			process.env.REACT_APP_EMAILJS_SERVICEID,
			process.env.REACT_APP_EMAILJS_USERID,
			process.env.REACT_APP_EMAILJS_RECEIVER
		);
		reset();
	};

	const sendMessage = (senderName, senderEmail, senderMessage, templateId, serviceId, userID) => {
		window.emailjs
			.send(
				serviceId,
				templateId,
				{
					from_email: senderEmail,
					from_name: senderName,
					message_html: senderMessage
				},
				userID
			)
			.then(() => {
				setSending(false);
			})
			.catch((err) => console.error('Failed to send message. Error: ', err));
	}
	return (
		<Layout>
			<Form onSubmit={handleSubmit(onSubmit)} register={register} sending={sending} />
		</Layout>
	);
}

const Layout = styled.div`
	width: 500px;
	max-width: 95%;
	margin: 0 auto;
`;
