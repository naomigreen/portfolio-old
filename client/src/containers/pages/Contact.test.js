import React from 'react';
import { render } from '@testing-library/react';
import Contact from './Contact';

describe('Contact component', () => {
	const { asFragment } = render(<Contact />);

	it('should render Contact component', () => {
		expect(asFragment()).toMatchSnapshot();
	});

});
