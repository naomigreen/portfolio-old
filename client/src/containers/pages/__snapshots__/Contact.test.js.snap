// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component should render Contact component 1`] = `
<DocumentFragment>
  <div
    class="sc-Axmtr jEDxkM"
  >
    <img
      alt=""
      class="sc-AxjAm bOyKAc"
      opacity="0"
      src="success.png"
      width="40px"
    />
    <form
      data-testid="form"
    >
      <input
        class="sc-AxhCb cHpGXh"
        name="name"
        placeholder="Name"
        type="text"
      />
      <input
        class="sc-AxhCb cHpGXh"
        name="email"
        placeholder="Email"
        type="email"
      />
      <textarea
        class="sc-AxhUy gYTJKi"
        name="message"
        placeholder="Message"
        type="text"
      />
      <button
        class="sc-AxgMl lnDpCC"
        data-testid="submit"
        type="submit"
      >
        Send message
      </button>
    </form>
    <div
      class="sc-AxheI jBPegG"
    >
      <a
        href="https://www.linkedin.com/in/naomi-prescod-green-3299868a/"
      >
        <img
          alt=""
          class="sc-AxiKw dgiEoI"
          src="linkedin.png"
        />
      </a>
      <a
        href="https://github.com/naomigreen"
      >
        <img
          alt=""
          class="sc-AxiKw dgiEoI"
          src="github.png"
        />
      </a>
      <a
        href="https://github.com/naomigreen/portfolio"
      >
        <img
          alt=""
          class="sc-AxiKw dgiEoI"
          src="code.png"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
